Add a section to introduce GradBucket class in ddp_comm_hooks.rst

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53253 Add a section to introduce GradBucket class in ddp_comm_hooks.rst**
* #53596 Add a description of GradBucket Python class

Since GradBucket class becomes public, mention this class in ddp_comm_hooks.rst.

Screenshot:
![Screen Shot 2021-03-09 at 1 42 03 AM](https://user-images.githubusercontent.com/8884619/110451202-d0a77b80-8078-11eb-9fb5-5cd015e406c9.png)

Differential Revision: [D26812210](https://our.internmc.facebook.com/intern/diff/D26812210/)

Commit:
Add a pointer to GradBucket class in ddp_comm_hooks.rst

Since GradBucket class becomes public, mention this class in ddp_comm_hooks.rst.

Screenshot: {F450211684}

Differential Revision: [D26812210](https://our.internmc.facebook.com/intern/diff/D26812210/)

[ghstack-poisoned]

pritamdamania87 commented on docs/source/ddp_comm_hooks.rst:
```diff
@@ -22,14 +22,16 @@ How to Use a Communication Hook?
 
 Default communication hooks are simple **stateless** hooks, so the input state
 in ``register_comm_hook`` is either a process group or ``None``.
+The input ``bucket`` is a :class:`torch.distributed.GradBucket` object,
```
Is the `GradBucket` class documented somewhere?

SciPioneer replied:
> Is the `GradBucket` class documented somewhere?

Similar to `register_comm_hook`, won't the definition be found in https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html?

Or do we need to provide a separate `autoclass:: torch.distributed.GradBucket`?

rohan-varma replied:
I think we may need the autoclass. Since `GradBucket` is now public, could you post what the docs look like and ensure `GradBucket` and its methods are appropriately rendered?

SciPioneer replied:
Added to the description of this PR.

rohan-varma reviewed (approved):
Thanks for documenting!

Commit:
Update on "Add a section to introduce GradBucket class in ddp_comm_hooks.rst"


Since GradBucket class becomes public, mention this class in ddp_comm_hooks.rst.

Screenshot:
![Screen Shot 2021-03-09 at 1 42 03 AM](https://user-images.githubusercontent.com/8884619/110451202-d0a77b80-8078-11eb-9fb5-5cd015e406c9.png)

Differential Revision: [D26812210](https://our.internmc.facebook.com/intern/diff/D26812210/)

[ghstack-poisoned]

Commit:
Update on "Add a section to introduce GradBucket class in ddp_comm_hooks.rst"


Since GradBucket class becomes public, mention this class in ddp_comm_hooks.rst.

Screenshot:
![Screen Shot 2021-03-09 at 1 42 03 AM](https://user-images.githubusercontent.com/8884619/110451202-d0a77b80-8078-11eb-9fb5-5cd015e406c9.png)

Differential Revision: [D26812210](https://our.internmc.facebook.com/intern/diff/D26812210/)

[ghstack-poisoned]
